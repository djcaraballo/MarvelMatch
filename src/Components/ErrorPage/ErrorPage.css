@import url('https://fonts.googleapis.com/css?family=Fjalla+One|Orbitron');

.error-wrapper {
  background: url('../../assets/Images/groot-404.jpg');
  background-size: cover;
  height: 100vh;
  width: 100vw;
}

.nav {
  text-align: left;
}

.home-link {
  margin: 1.5em;
  font-size: 2em;
}

.warning-container {
  text-align: left;
  width: 35%;
  padding: 1em;
  margin-left: 1.5em
}

h1 {
  font-size: 4.5em;
  font-family: 'Fjalla One', sans-serif;
  color: #fff;
}

.translation {
  font-family: "Orbitron", sans-serif;
  line-height: 1.5em;
  font-size: 1.5em;
  letter-spacing: 1px;
}