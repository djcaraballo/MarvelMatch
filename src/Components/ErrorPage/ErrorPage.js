import React from 'react'
import { NavLink } from 'react-router-dom'
import './ErrorPage.css'

const ErrorPage = () => {
  return (
    <div className="error-wrapper">
      <div className='warning-container'>
        <h1>404: I AM GROOT!</h1>
        <p className='translation'>Translation: Not quite sure where you were headed, friend, but this path leads to Knowhere and you'll never survive the trip!</p>
      </div>
      <NavLink to='/quiz-land' className='error-nav'>
        <p className='home-link'>Back to Safety!</p>
      </NavLink>
    </div>
  )
}

export default ErrorPage