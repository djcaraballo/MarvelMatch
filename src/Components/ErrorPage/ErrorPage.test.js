import React from 'react'
import ErrorPage from './ErrorPage'
import { shallow } from 'enzyme'

describe('ErrorPage Component', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(<ErrorPage />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})