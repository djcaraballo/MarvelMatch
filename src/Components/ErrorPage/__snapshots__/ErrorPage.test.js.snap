// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorPage Component should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ErrorPage />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <NavLink
          activeClassName="active"
          aria-current="page"
          className="nav"
          to="/Home"
        >
          <p
            className="home-link"
          >
            Back To Home
          </p>
        </NavLink>,
        <div
          className="warning-container"
        >
          <h1>
            404: I AM GROOT!
          </h1>
          <p
            className="translation"
          >
            Translation: Not quite sure where you were headed, friend, but this path leads to Knowhere and you'll never survive the trip!
          </p>
        </div>,
      ],
      "className": "error-wrapper",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "aria-current": "page",
          "children": <p
            className="home-link"
          >
            Back To Home
          </p>,
          "className": "nav",
          "to": "/Home",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Back To Home",
            "className": "home-link",
          },
          "ref": null,
          "rendered": "Back To Home",
          "type": "p",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              404: I AM GROOT!
            </h1>,
            <p
              className="translation"
            >
              Translation: Not quite sure where you were headed, friend, but this path leads to Knowhere and you'll never survive the trip!
            </p>,
          ],
          "className": "warning-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "404: I AM GROOT!",
            },
            "ref": null,
            "rendered": "404: I AM GROOT!",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Translation: Not quite sure where you were headed, friend, but this path leads to Knowhere and you'll never survive the trip!",
              "className": "translation",
            },
            "ref": null,
            "rendered": "Translation: Not quite sure where you were headed, friend, but this path leads to Knowhere and you'll never survive the trip!",
            "type": "p",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <NavLink
            activeClassName="active"
            aria-current="page"
            className="nav"
            to="/Home"
          >
            <p
              className="home-link"
            >
              Back To Home
            </p>
          </NavLink>,
          <div
            className="warning-container"
          >
            <h1>
              404: I AM GROOT!
            </h1>
            <p
              className="translation"
            >
              Translation: Not quite sure where you were headed, friend, but this path leads to Knowhere and you'll never survive the trip!
            </p>
          </div>,
        ],
        "className": "error-wrapper",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": <p
              className="home-link"
            >
              Back To Home
            </p>,
            "className": "nav",
            "to": "/Home",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Back To Home",
              "className": "home-link",
            },
            "ref": null,
            "rendered": "Back To Home",
            "type": "p",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1>
                404: I AM GROOT!
              </h1>,
              <p
                className="translation"
              >
                Translation: Not quite sure where you were headed, friend, but this path leads to Knowhere and you'll never survive the trip!
              </p>,
            ],
            "className": "warning-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "404: I AM GROOT!",
              },
              "ref": null,
              "rendered": "404: I AM GROOT!",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Translation: Not quite sure where you were headed, friend, but this path leads to Knowhere and you'll never survive the trip!",
                "className": "translation",
              },
              "ref": null,
              "rendered": "Translation: Not quite sure where you were headed, friend, but this path leads to Knowhere and you'll never survive the trip!",
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
